src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -1651,6 +1656,19 @@ parser::parse_and_visit_function_parameters(
     });
     return function_parameter_parse_result::missing_parameters_ignore_body;
 
+    // function async f() {}  // Invalid. Should be async function f() {}
+  case token_type::identifier:
+    if (name->string_view() ==
+        u8"async"_sv) {  // In this context async is of type identifier and not
+                         // kw_async so we need to look at the name.
```

Comment:
Nit: This comment is misleading. The real issue is that we don't have the `token` object. I think we don't need a comment here, or if there is one, it should be something like:

> TODO: Make parse_and_visit_function_parameters accept a token instead of a source_code_span so we can compare the token type instead of strings.